### PR TITLE
Added the mult operation to Basis Class

### PR DESCRIPTION
**Pull request by manso92**

### CODE REVIEW

File: fda/basis.py [hunk 1]
```diff
@@ -252,12 +252,238 @@ def penalty(self, derivative_degree=None, coefficients=None):
         """
         pass
 
+    def __mul__(self, other):
+        """
+            Multiplication for FDataBasis object.
+        """
+
+        if self.domain_range == other.domain_range:
```
[the remark this answers is not in the document]
Comment (manso92) [reply fda/basis.py]: Yes, a last change mistake

File: fda/basis.py [hunk 2]
```diff
@@ -252,12 +252,238 @@ def penalty(self, derivative_degree=None, coefficients=None):
...
+            raise ValueError("Ranges are not equal.")
+
+        if isinstance(self, Constant) and isinstance(other, Constant):
+            return Constant(self.domain_range)
```
[the remark this answers is not in the document]
Comment (manso92) [reply fda/basis.py]: Yes

File: fda/basis.py [hunk 3]
```diff
@@ -430,13 +557,24 @@ def penalty(self, derivative_degree=None, coefficients=None):
...
                     penalty_matrix[jbasis, ibasis] = penalty_matrix[ibasis,
                                                                     jbasis]
 
         return penalty_matrix
 
+    def __mul__(self, other):
+        """Multiplication of a Monomial Basis with other Basis"""
+        if self.domain_range != other.domain_range:
+            raise ValueError("Ranges are not equal.")
+
```
Comment (vnmabus) [inline fda/basis.py]: Can this not be implemented when `other` is also Monomial?

**Comment by manso92** [comment]
Fixed all changes to pass the pull request

**Pull request merged**